cmd/install/assets/hypershift_operator.go

```diff
@@ -127,26 +127,28 @@ func (o HyperShiftOperatorDeployment) Build() *appsv1.Deployment {
 		},
 	}
 
-	args = append(args,
-		"--oidc-storage-provider-s3-bucket-name="+o.OIDCBucketName,
-		"--oidc-storage-provider-s3-region="+o.OIDCBucketRegion,
-		"--oidc-storage-provider-s3-credentials=/etc/oidc-storage-provider-s3-creds/"+o.OIDCStorageProviderS3SecretKey,
-	)
-	oidcVolumeMount = []corev1.VolumeMount{
-		{
-			Name:      "oidc-storage-provider-s3-creds",
-			MountPath: "/etc/oidc-storage-provider-s3-creds",
-		},
-	}
-	oidcVolumeCred = []corev1.Volume{
-		{
-			Name: "oidc-storage-provider-s3-creds",
-			VolumeSource: corev1.VolumeSource{
-				Secret: &corev1.SecretVolumeSource{
-					SecretName: o.OIDCStorageProviderS3Secret.Name,
+	if o.OIDCStorageProviderS3Secret != nil {
```

Comment:
How about a platform check instead?

Reply:
The platform type is not passed to this Build function
I think this is good indication at this point
Is there any other parameter you suggest to use here instead?

Reply:
If parts of the resource graph to create are platform dependent, then maybe `Build()` should accept a platform type. The various OIDC flag checks are effectively an implicit platform check and are obfuscating the fact that install now has platform-isms.

What if the `install` command had a `--platform` flag (defaulting to `aws`) which is passed through to make the platform specific logic explicit?

cc @sjenning @alvaroaleman @enxebre

Reply:
We need to consider IBMCloud in this as well (follows similar path to none right now)

Reply:
We also need to consider the defaults in the deployment now as I don't think this will change that

Reply:
Sample test code
```
package assets

import (
	"github.com/onsi/gomega"
	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"testing"
)

func TestHyperShiftOperatorDeployment_Build_OIDC(t *testing.T) {
	testNamespace := "hypershift"
	testOperatorImage := "myimage"
	tests := map[string]struct {
		inputBuildParameters HyperShiftOperatorDeployment
		expectedVolumeMounts []corev1.VolumeMount
		expectedVolumes      []corev1.Volume
	}{
		"empty oidc paramaters result in no volume mounts": {
			inputBuildParameters: HyperShiftOperatorDeployment{
				Namespace: &corev1.Namespace{
					ObjectMeta: metav1.ObjectMeta{
						Name: testNamespace,
					},
				},
				OperatorImage: testOperatorImage,
				ServiceAccount: &corev1.ServiceAccount{
					ObjectMeta: metav1.ObjectMeta{
						Name: "hypershift",
					},
				},
				Replicas:        3,
				PrivatePlatform: string(hyperv1.NonePlatform),
			},
			expectedVolumeMounts: nil,
			expectedVolumes:      nil,
		},
	}
	for name, test := range tests {
		t.Run(name, func(t *testing.T) {
			g := gomega.NewGomegaWithT(t)
			deployment := test.inputBuildParameters.Build()
			g.Expect(deployment.Spec.Template.Spec.Volumes).To(gomega.BeEquivalentTo(test.expectedVolumes))
			g.Expect(deployment.Spec.Template.Spec.Containers[0].VolumeMounts).To(gomega.BeEquivalentTo(test.expectedVolumeMounts))
		})
	}
}
```